### PR TITLE
More verbose agg debugging; checks pairing during circuit generation

### PR DESCRIPTION
**Pull request by zhenfeizhang**
### Description

1. previously, when sanity check failed, the code panics with an error message that `pub(crate) fn assert_equal<F: Field>()` failed. This is not very helpful for debugging as multiple checks use this same `assert_equal()` function.

this PR dumps more information when assertion failed during sanity checks.

2. during circuit generation, the input accumulators and output accumulator are both checked for correctness

~This is a breaking change in API cc @silathdiir~ 

### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Rationale

[_design decisions and extended information_]

### CODE REVIEW

File: aggregator/src/aggregation/circuit.rs [hunk 1]
```diff
@@ -63,6 +66,8 @@ impl AggregationCircuit {
         snarks_with_padding: &[Snark],
         rng: impl Rng + Send,
         batch_hash: BatchHash,
+        g2: &G2Affine,
```
Comment (zhenfeizhang) [inline aggregator/src/aggregation/circuit.rs]: breaking changes in circuit API @silathdiir 

`g2` and `s_g2` are from parameters

File: aggregator/src/compression/circuit.rs [hunk 2]
```diff
@@ -174,6 +177,8 @@ impl CompressionCircuit {
         snark: Snark,
         has_accumulator: bool,
         rng: impl Rng + Send,
+        g2: &G2Affine,
```
Comment (zhenfeizhang) [inline aggregator/src/compression/circuit.rs]: breaking changes in circuit API @silathdiir

`g2` and `s_g2` are from parameters

File: aggregator/src/aggregation/circuit.rs [hunk 3]
```diff
@@ -94,8 +99,18 @@ impl AggregationCircuit {
...
         let KzgAccumulator::<G1Affine, NativeLoader> { lhs, rhs } = accumulator;
+
+        // sanity check on the accumulator
+        {
+            let left = Bn256::pairing(&lhs, g2);
+            let right = Bn256::pairing(&rhs, s_g2);
+            log::trace!("aggregation circuit acc check: left {:?}", left);
+            log::trace!("aggregation circuit acc check: right {:?}", right);
+            assert_eq!(left, right, "accumulator check failed");
+        }
```
Comment (lispc) [inline aggregator/src/aggregation/circuit.rs:119]: what about return err?  it is not something really rare.

File: aggregator/src/compression/circuit.rs [hunk 4]
```diff
@@ -191,6 +196,16 @@ impl CompressionCircuit {
...
         // as specified in CircuitExt::accumulator_indices()
         let KzgAccumulator::<G1Affine, NativeLoader> { lhs, rhs } = accumulator;
+
+        // sanity check on the accumulator
+        {
+            let left = Bn256::pairing(&lhs, g2);
+            let right = Bn256::pairing(&rhs, s_g2);
+            log::trace!("compression circuit acc check: left {:?}", left);
+            log::trace!("compression circuit acc check: right {:?}", right);
+            assert_eq!(left, right, "accumulator check failed");
```
Comment (lispc) [inline aggregator/src/compression/circuit.rs]: ditto

File: aggregator/src/util.rs [hunk 5]
```diff
@@ -99,12 +99,16 @@ pub(crate) fn get_indices(preimages: &[Vec<u8>]) -> (Vec<usize>, Vec<usize>) {
...
+    a: &AssignedCell<F, F>,
+    b: &AssignedCell<F, F>,
+    description: &str,
+) {
     let mut t1 = F::default();
     let mut t2 = F::default();
     a.value().map(|f| t1 = *f);
     b.value().map(|f| t2 = *f);
-    assert_eq!(t1, t2)
+    assert_eq!(t1, t2, "{description}",)
```
Comment (lispc) [inline aggregator/src/util.rs:111]: does this work?    I am not sure, i usually will write assert_eq!(t1, t2, format!("{description}"));

Comment (zhenfeizhang) [reply aggregator/src/util.rs:111]: from clippy:
```
   --> aggregator/src/util.rs:130:9
    |
130 |         assert_eq!(t1, t2, "{}",description)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args
help: change this to
    |
130 -         assert_eq!(t1, t2, "{}",description)
130 +         assert_eq!(t1, t2, "{description}")
```

**Comment by zhenfeizhang** [comment]
> This is a breaking change in API cc @silathdiir

i was wrong... I managed to keep the same API. so no changes needed

**Pull request merged**